### ISSUE TITLE
Support MC 1.21.4

### ISSUE THREAD

**Issue by Chi-tao** [MinecraftFreecam/Freecam#259]
### Description of the issue

Game works perfectly fine until I try and use freecam. I hit the freecam button, I go into the cam mode and then my game crashes. Everything is updated and it still doesn't work with this and fabricapi being the only mods installed. 

### Expected behavior

I expect to not crash when using freecam

### Steps to reproduce

1. Open a singleplayer world
2. Enable Freecam by pressing `F4`
3. crash

### Log output

```shell
[22:43:47] [main/INFO]: Loading Minecraft 1.21.4 with Fabric Loader 0.16.9
[22:43:47] [main/INFO]: Loading 54 mods:
	- fabric-api 0.111.0+1.21.4
	   |-- fabric-api-base 0.4.53+453d4f9104
	   |-- fabric-api-lookup-api-v1 1.6.84+7feeb73304
	   |-- fabric-biome-api-v1 15.0.4+466a140b04
	   |-- fabric-block-api-v1 1.0.31+7feeb73304
	   |-- fabric-block-view-api-v2 1.0.19+7feeb73304
	   |-- fabric-blockrenderlayer-v1 2.0.7+7feeb73304
	   |-- fabric-client-tags-api-v1 1.1.27+7feeb73304
	   |-- fabric-command-api-v1 1.2.61+f71b366f04
	   |-- fabric-command-api-v2 2.2.40+e496eb1504
	   |-- fabric-commands-v0 0.2.78+df3654b304
	   |-- fabric-content-registries-v0 9.1.12+7feeb73304
	   |-- fabric-convention-tags-v1 2.1.6+7f945d5b04
	   |-- fabric-convention-tags-v2 2.9.6+466a140b04
	   |-- fabric-crash-report-info-v1 0.3.6+7feeb73304
	   |-- fabric-data-attachment-api-v1 1.4.3+9aea556b04
	   |-- fabric-data-generation-api-v1 22.1.2+466a140b04
	   |-- fabric-dimensions-v1 4.0.10+7feeb73304
	   |-- fabric-entity-events-v1 2.0.11+60fccc7604
	   |-- fabric-events-interaction-v0 4.0.1+a4eebcf004
	   |-- fabric-game-rule-api-v1 1.0.62+7feeb73304
	   |-- fabric-item-api-v1 11.1.12+203e6b2304
	   |-- fabric-item-group-api-v1 4.1.21+7feeb73304
	   |-- fabric-key-binding-api-v1 1.0.56+7feeb73304
	   |-- fabric-keybindings-v0 0.2.54+df3654b304
	   |-- fabric-lifecycle-events-v1 2.5.3+bf2a60eb04
	   |-- fabric-loot-api-v2 3.0.31+3f89f5a504
	   |-- fabric-loot-api-v3 1.0.19+203e6b2304
	   |-- fabric-message-api-v1 6.0.24+7feeb73304
	   |-- fabric-model-loading-api-v1 4.1.0+7b6b225304
	   |-- fabric-networking-api-v1 4.3.8+cc0fa2fe04
	   |-- fabric-object-builder-api-v1 18.0.5+e604fe7f04
	   |-- fabric-particles-v1 4.0.13+7feeb73304
	   |-- fabric-recipe-api-v1 8.0.3+fa62a02304
	   |-- fabric-registry-sync-v0 6.0.1+cc0fa2fe04
	   |-- fabric-renderer-api-v1 5.0.1+84404cdd04
	   |-- fabric-renderer-indigo 2.0.1+84404cdd04
	   |-- fabric-rendering-data-attachment-v1 0.3.57+73761d2e04
	   |-- fabric-rendering-fluids-v1 3.1.18+7feeb73304
	   |-- fabric-rendering-v1 10.1.0+7e7568bf04
	   |-- fabric-resource-conditions-api-v1 5.0.13+203e6b2304
	   |-- fabric-resource-loader-v0 3.0.10+203e6b2304
	   |-- fabric-screen-api-v1 2.0.37+7feeb73304
	   |-- fabric-screen-handler-api-v1 1.3.106+7feeb73304
	   |-- fabric-sound-api-v1 1.0.32+7feeb73304
	   |-- fabric-transfer-api-v1 5.4.6+1885ad7404
	   \-- fabric-transitive-access-wideners-v1 6.3.2+56e78b9b04
	- fabricloader 0.16.9
	   \-- mixinextras 0.4.1
	- freecam 1.3.1+mc1.21.3
	   \-- cloth-config 16.0.141
	        \-- cloth-basic-math 0.6.1
	- java 21
	- minecraft 1.21.4
[22:43:47] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.7 Source=file:/C:/Users/okorc/AppData/Roaming/.minecraft/libraries/net/fabricmc/sponge-mixin/0.15.4+mixin.0.8.7/sponge-mixin-0.15.4+mixin.0.8.7.jar Service=Knot/Fabric Env=CLIENT
[22:43:47] [main/INFO]: Compatibility level set to JAVA_21
[22:43:48] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.4.1).
[22:43:49] [Datafixer Bootstrap/INFO]: 243 Datafixer optimizations took 429 milliseconds
[22:43:49] [main/WARN]: Invalid registry value type detected for PerfOS counters. Should be REG_DWORD. Ignoring: HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\PerfOS\Performance\Disable Performance Counters.
[22:43:54] [Render thread/INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, name=PROD]
[22:43:54] [Render thread/INFO]: Setting user: Chi_Tao
[22:43:54] [Render thread/INFO]: [Indigo] Registering Indigo renderer!
[22:43:54] [Render thread/INFO]: Backend library: LWJGL version 3.3.3-snapshot
[22:43:55] [Render thread/INFO]: Reloading ResourceManager: vanilla, fabric, cloth-config, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-blockrenderlayer-v1, fabric-client-tags-api-v1, fabric-command-api-v1, fabric-command-api-v2, fabric-commands-v0, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-keybindings-v0, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-sound-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, freecam
[22:43:55] [Worker-Main-10/INFO]: Found unifont_jp_patch-16.0.01.hex, loading
[22:43:55] [Worker-Main-8/INFO]: Found unifont_all_no_pua-16.0.01.hex, loading
[22:43:57] [Render thread/INFO]: OpenAL initialized on device OpenAL Soft on Lautsprecher (PRO X Wireless Gaming Headset)
[22:43:57] [Render thread/INFO]: Sound engine started
[22:43:57] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/blocks.png-atlas
[22:43:57] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/signs.png-atlas
[22:43:57] [Render thread/INFO]: Created: 512x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[22:43:57] [Render thread/INFO]: Created: 512x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[22:43:57] [Render thread/INFO]: Created: 2048x1024x4 minecraft:textures/atlas/armor_trims.png-atlas
[22:43:57] [Render thread/INFO]: Created: 128x64x4 minecraft:textures/atlas/decorated_pot.png-atlas
[22:43:57] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[22:43:57] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[22:43:57] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[22:43:57] [Render thread/INFO]: Created: 64x64x0 minecraft:textures/atlas/map_decorations.png-atlas
[22:43:57] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/particles.png-atlas
[22:43:57] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/paintings.png-atlas
[22:43:57] [Render thread/INFO]: Created: 256x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[22:43:57] [Render thread/INFO]: Created: 1024x512x0 minecraft:textures/atlas/gui.png-atlas
[22:44:04] [Render thread/INFO]: Loaded 1370 recipes
[22:44:04] [Render thread/INFO]: Loaded 1481 advancements
[22:44:05] [Render thread/INFO]: Applied 0 biome modifications to 0 of 65 new biomes in 1.272 ms
[22:44:05] [Server thread/INFO]: Starting integrated minecraft server version 1.21.4
[22:44:05] [Server thread/INFO]: Generating keypair
[22:44:05] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[22:44:06] [Render thread/INFO]: Preparing spawn area: 0%
[22:44:06] [Render thread/INFO]: Preparing spawn area: 0%
[22:44:06] [Render thread/INFO]: Time elapsed: 812 ms
[22:44:06] [Server thread/INFO]: Changing view distance to 12, from 10
[22:44:06] [Server thread/INFO]: Changing simulation distance to 12, from 0
[22:44:06] [Server thread/INFO]: Chi_Tao[local:E:153b66f6] logged in with entity id 10 at (-105.34815527663677, 72.0, 1097.5805841412557)
[22:44:06] [Server thread/INFO]: Chi_Tao joined the game
[22:44:07] [Render thread/INFO]: Loaded 612 advancements
[22:44:11] [Render thread/ERROR]: Reported exception thrown!
net.minecraft.class_148: Ticking entity
	at knot/net.minecraft.class_1937.method_18472(class_1937.java:509) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_638.method_32124(class_638.java:263) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_5574.method_31791(class_5574.java:54) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_638.method_18116(class_638.java:259) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1574(class_310.java:1879) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1523(class_310.java:1302) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1514(class_310.java:922) [client-intermediary.jar:?]
	at knot/net.minecraft.client.main.Main.main(Main.java:267) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.9.jar:?]
Caused by: java.lang.NoSuchMethodError: 'void net.minecraft.class_744.method_3129(boolean, float)'
	at knot/net.xolt.freecam.util.Motion.doMotion(Motion.java:19) ~[freecam-fabric-modrinth-1.3.1+mc1.21.3.jar:?]
	at knot/net.xolt.freecam.util.FreeCamera.method_6007(FreeCamera.java:236) ~[freecam-fabric-modrinth-1.3.1+mc1.21.3.jar:?]
	at knot/net.minecraft.class_1309.method_5773(class_1309.java:2606) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1657.method_5773(class_1657.java:321) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_742.method_5773(class_742.java:62) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_746.method_5773(class_746.java:224) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_638.method_18646(class_638.java:284) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1937.method_18472(class_1937.java:502) ~[client-intermediary.jar:?]
	... 10 more
[22:44:12] [Server thread/INFO]: Stopping server
[22:44:12] [Server thread/INFO]: Saving players
[22:44:12] [Server thread/INFO]: Saving worlds
[22:44:13] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Farms]'/minecraft:overworld
[22:44:13] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Farms]'/minecraft:the_end
[22:44:13] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Farms]'/minecraft:the_nether
[22:44:13] [Server thread/INFO]: ThreadedAnvilChunkStorage (Farms): All chunks are saved
[22:44:13] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[22:44:13] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[22:44:13] [Server thread/INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
[22:44:13] [Sound engine/ERROR]: Stop: Invalid name parameter.
```

### Crash report

```shell
---- Minecraft Crash Report ----
// Hey, that tickles! Hehehe!

Time: 2024-12-05 22:44:13
Description: Ticking entity

java.lang.NoSuchMethodError: 'void net.minecraft.class_744.method_3129(boolean, float)'
	at knot//net.xolt.freecam.util.Motion.doMotion(Motion.java:19)
	at knot//net.xolt.freecam.util.FreeCamera.method_6007(FreeCamera.java:236)
	at knot//net.minecraft.class_1309.method_5773(class_1309.java:2606)
	at knot//net.minecraft.class_1657.method_5773(class_1657.java:321)
	at knot//net.minecraft.class_742.method_5773(class_742.java:62)
	at knot//net.minecraft.class_746.method_5773(class_746.java:224)
	at knot//net.minecraft.class_638.method_18646(class_638.java:284)
	at knot//net.minecraft.class_1937.method_18472(class_1937.java:502)
	at knot//net.minecraft.class_638.method_32124(class_638.java:263)
	at knot//net.minecraft.class_5574.method_31791(class_5574.java:54)
	at knot//net.minecraft.class_638.method_18116(class_638.java:259)
	at knot//net.minecraft.class_310.method_1574(class_310.java:1879)
	at knot//net.minecraft.class_310.method_1523(class_310.java:1302)
	at knot//net.minecraft.class_310.method_1514(class_310.java:922)
	at knot//net.minecraft.client.main.Main.main(Main.java:267)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at knot//net.xolt.freecam.util.Motion.doMotion(Motion.java:19)
	at knot//net.xolt.freecam.util.FreeCamera.method_6007(FreeCamera.java:236)
	at knot//net.minecraft.class_1309.method_5773(class_1309.java:2606)
	at knot//net.minecraft.class_1657.method_5773(class_1657.java:321)
	at knot//net.minecraft.class_742.method_5773(class_742.java:62)
	at knot//net.minecraft.class_746.method_5773(class_746.java:224)
	at knot//net.minecraft.class_638.method_18646(class_638.java:284)
	at knot//net.minecraft.class_1937.method_18472(class_1937.java:502)
	at knot//net.minecraft.class_638.method_32124(class_638.java:263)
	at knot//net.minecraft.class_5574.method_31791(class_5574.java:54)
	at knot//net.minecraft.class_638.method_18116(class_638.java:259)

-- Entity being ticked --
Details:
	Entity Type: minecraft:player (net.xolt.freecam.util.FreeCamera)
	Entity ID: -420
	Entity Name: FreeCamera
	Entity's Exact location: -105.35, 73.22, 1097.58
	Entity's Block location: World: (-106,73,1097), Section: (at 6,9,9 in -7,4,68; chunk contains blocks -112,-64,1088 to -97,319,1103), Region: (-1,2; contains chunks -32,64 to -1,95, blocks -512,-64,1024 to -1,319,1535)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null
Stacktrace:
	at knot//net.minecraft.class_1937.method_18472(class_1937.java:502)
	at knot//net.minecraft.class_638.method_32124(class_638.java:263)
	at knot//net.minecraft.class_5574.method_31791(class_5574.java:54)
	at knot//net.minecraft.class_638.method_18116(class_638.java:259)
	at knot//net.minecraft.class_310.method_1574(class_310.java:1879)
	at knot//net.minecraft.class_310.method_1523(class_310.java:1302)
	at knot//net.minecraft.class_310.method_1514(class_310.java:922)
	at knot//net.minecraft.client.main.Main.main(Main.java:267)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Uptime --
Details:
	JVM uptime: 26.838s
	Wall uptime: 18.841s
	High-res time: 18.101s
	Client ticks: 275 ticks / 13.750s
Stacktrace:
	at knot//net.minecraft.class_310.method_1587(class_310.java:2371)
	at knot//net.minecraft.class_310.method_54580(class_310.java:991)
	at knot//net.minecraft.class_310.method_1514(class_310.java:943)
	at knot//net.minecraft.client.main.Main.main(Main.java:267)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 2 total; [class_746['Chi_Tao'/10, l='ClientLevel', x=-105.35, y=72.00, z=1097.58], FreeCamera['FreeCamera'/-420, l='ClientLevel', x=-105.35, y=73.22, z=1097.58]]
	Chunk stats: 961, 548
	Level dimension: minecraft:overworld
	Level spawn location: World: (-144,69,448), Section: (at 0,5,0 in -9,4,28; chunk contains blocks -144,-64,448 to -129,319,463), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Level time: 4372578 game time, 13000 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
	Tracked entity count: 408

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: vanilla, fabric, cloth-config, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-blockrenderlayer-v1, fabric-client-tags-api-v1, fabric-command-api-v1, fabric-command-api-v2, fabric-commands-v0, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-keybindings-v0, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-sound-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, freecam

-- System Details --
Details:
	Minecraft Version: 1.21.4
	Minecraft Version ID: 1.21.4
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 21.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 467794240 bytes (446 MiB) / 1073741824 bytes (1024 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 16
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 7 5800X3D 8-Core Processor           
	Identifier: AuthenticAMD Family 25 Model 33 Stepping 2
	Microarchitecture: Zen 3
	Frequency (GHz): 3.39
	Number of physical packages: 1
	Number of physical CPUs: 8
	Number of logical CPUs: 16
	Graphics card #0 name: NVIDIA GeForce RTX 3060 Ti
	Graphics card #0 vendor: NVIDIA
	Graphics card #0 VRAM (MiB): 8192.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 32.0.15.6614
	Memory slot #0 capacity (MiB): 16384.00
	Memory slot #0 clockSpeed (GHz): 3.20
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MiB): 16384.00
	Memory slot #1 clockSpeed (GHz): 3.20
	Memory slot #1 type: DDR4
	Virtual memory max (MiB): 38025.15
	Virtual memory used (MiB): 19698.77
	Swap memory total (MiB): 5376.00
	Swap memory used (MiB): 129.19
	Space in storage for jna.tmpdir (MiB): available: 329363.19, total: 953204.88
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): available: 329363.19, total: 953204.88
	Space in storage for io.netty.native.workdir (MiB): available: 329363.19, total: 953204.88
	Space in storage for java.io.tmpdir (MiB): available: 329363.19, total: 953204.88
	Space in storage for workdir (MiB): available: 329363.19, total: 953204.88
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		fabric-api: Fabric API 0.111.0+1.21.4
			fabric-api-base: Fabric API Base 0.4.53+453d4f9104
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.84+7feeb73304
			fabric-biome-api-v1: Fabric Biome API (v1) 15.0.4+466a140b04
			fabric-block-api-v1: Fabric Block API (v1) 1.0.31+7feeb73304
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.19+7feeb73304
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 2.0.7+7feeb73304
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.27+7feeb73304
			fabric-command-api-v1: Fabric Command API (v1) 1.2.61+f71b366f04
			fabric-command-api-v2: Fabric Command API (v2) 2.2.40+e496eb1504
			fabric-commands-v0: Fabric Commands (v0) 0.2.78+df3654b304
			fabric-content-registries-v0: Fabric Content Registries (v0) 9.1.12+7feeb73304
			fabric-convention-tags-v1: Fabric Convention Tags 2.1.6+7f945d5b04
			fabric-convention-tags-v2: Fabric Convention Tags (v2) 2.9.6+466a140b04
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.3.6+7feeb73304
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.4.3+9aea556b04
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 22.1.2+466a140b04
			fabric-dimensions-v1: Fabric Dimensions API (v1) 4.0.10+7feeb73304
			fabric-entity-events-v1: Fabric Entity Events (v1) 2.0.11+60fccc7604
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 4.0.1+a4eebcf004
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.62+7feeb73304
			fabric-item-api-v1: Fabric Item API (v1) 11.1.12+203e6b2304
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.1.21+7feeb73304
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.56+7feeb73304
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.54+df3654b304
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.5.3+bf2a60eb04
			fabric-loot-api-v2: Fabric Loot API (v2) 3.0.31+3f89f5a504
			fabric-loot-api-v3: Fabric Loot API (v3) 1.0.19+203e6b2304
			fabric-message-api-v1: Fabric Message API (v1) 6.0.24+7feeb73304
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 4.1.0+7b6b225304
			fabric-networking-api-v1: Fabric Networking API (v1) 4.3.8+cc0fa2fe04
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 18.0.5+e604fe7f04
			fabric-particles-v1: Fabric Particles (v1) 4.0.13+7feeb73304
			fabric-recipe-api-v1: Fabric Recipe API (v1) 8.0.3+fa62a02304
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 6.0.1+cc0fa2fe04
			fabric-renderer-api-v1: Fabric Renderer API (v1) 5.0.1+84404cdd04
			fabric-renderer-indigo: Fabric Renderer - Indigo 2.0.1+84404cdd04
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.57+73761d2e04
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.1.18+7feeb73304
			fabric-rendering-v1: Fabric Rendering (v1) 10.1.0+7e7568bf04
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 5.0.13+203e6b2304
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 3.0.10+203e6b2304
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.37+7feeb73304
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.106+7feeb73304
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.32+7feeb73304
			fabric-transfer-api-v1: Fabric Transfer API (v1) 5.4.6+1885ad7404
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 6.3.2+56e78b9b04
		fabricloader: Fabric Loader 0.16.9
			mixinextras: MixinExtras 0.4.1
		freecam: Freecam (Modrinth Edition) 1.3.1+mc1.21.3
			cloth-config: Cloth Config v16 16.0.141
				cloth-basic-math: cloth-basic-math 0.6.1
		java: OpenJDK 64-Bit Server VM 21
		minecraft: Minecraft 1.21.4
	Launched Version: fabric-loader-0.16.9-1.21.4
	Launcher name: minecraft-launcher
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: NVIDIA GeForce RTX 3060 Ti/PCIe/SSE2 GL version 3.2.0 NVIDIA 566.14, NVIDIA Corporation
	Window size: 1920x1080
	GFLW Platform: win32
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Is Modded: Definitely; Client brand changed to 'fabric'; Server brand changed to 'fabric'
	Universe: 400921fb54442d18
	Type: Integrated Server (map_client.txt)
	Graphics mode: fancy
	Render Distance: 12/12 chunks
	Resource Packs: vanilla, fabric, cloth-config, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-blockrenderlayer-v1, fabric-client-tags-api-v1, fabric-command-api-v1, fabric-command-api-v2, fabric-commands-v0, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-keybindings-v0, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-sound-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, freecam
	Current Language: en_us
	Locale: de_DE
	System encoding: Cp1252
	File encoding: UTF-8
	CPU: 16x AMD Ryzen 7 5800X3D 8-Core Processor 
	Server Running: true
	Player Count: 1 / 8; [class_3222['Chi_Tao'/10, l='ServerLevel[Farms]', x=-105.35, y=72.00, z=1097.58]]
	Active Data Packs: vanilla, fabric, fabric-convention-tags-v2
	Available Data Packs: fabric, fabric-convention-tags-v2, minecart_improvements, redstone_experiments, trade_rebalance, vanilla
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Stable
	World Seed: 4211257627163461106
	Suppressed Exceptions: ~~NONE~~
```

### How prevalent is the issue?

every time I press the key

### Freecam version

1.3.1+mc1.21.3

### Are you using "Modrinth Edition"?

Yes

### Minecraft version

1.21.4

### Modding platform

- [X] Fabric
- [ ] Neoforge
- [ ] Forge

**Comment by hashalite** [comment]
1.21.4 is not yet supported. However, there are beta builds available [here](https://github.com/MinecraftFreecam/Freecam/actions/runs/12149670068). The config gui currently doesn't work, but if your config is already set up it should be functional otherwise.

**Comment by MattSturgeon** [comment]
Let's keep this open to track 1.21.4 support

**Comment by MattSturgeon** [comment]
Fixed by #262, released as https://github.com/MinecraftFreecam/Freecam/releases/tag/v1.3.2
